File: core/src/main/java/heast/core/network/packets/c2s/auth/VerifyC2SPacket.java
package heast.core.network.packets.c2s.auth;

import heast.core.network.Packet;
import heast.core.network.PacketBuf;
import heast.core.network.listeners.ServerAuthListener;

public final class VerifyC2SPacket implements Packet<ServerAuthListener> {

    private final String code;

    public VerifyC2SPacket(String code) {
        this.code = code;
    }

    public VerifyC2SPacket(PacketBuf buf) {
        this.code = buf.readString();
    }

    @Override
    public void write(PacketBuf buf) {
        buf.writeString(code);
    }

    @Override
    public void apply(ServerAuthListener listener) {
        listener.onVerify(this);
    }

    public String getCode() {
        return code;
    }
}

File: Core/Network/Packets/C2S/VerifyC2SPacket.cs
using Core.Network.Listeners;

namespace Core.Network.Packets.C2S;

public class VerifyC2SPacket : AbstractPacket
{
    public VerifyC2SPacket(string code)
    {
        Code = code;
    }

    public VerifyC2SPacket(PacketBuf buf)
    {
        Code = buf.ReadString();
    }

    public string Code { get; }

    public override void Write(PacketBuf buf)
    {
        buf.WriteString(Code);
    }

    public override void Apply(IPacketListener listener)
    {
        if (listener is IServerAuthListener authListener)
        {
            authListener.OnVerify(this);
        }
    }
}